File: 백준/Gold/1715. 카드 정렬하기/카드 정렬하기.cc
#include <stdio.h>
#include <iostream>
#include <queue>
using namespace std;

int main(void) {
	int n, in, in2, res = 0;
	priority_queue<int,deque<int>,greater<int>> q;
	scanf("%d", &n);
	while (n--) {
		scanf("%d", &in);
		q.push(in);
	}
	while (q.size() != 1) {
		in = q.top();
		q.pop();
		in2 = q.top();
		q.pop();
		res += in + in2;
		q.push(in + in2);
	}
	cout << res;
	return 0;

}

File: 백준/Gold/1715. 카드 정렬하기/카드 정렬하기.py
from queue import PriorityQueue

pq = PriorityQueue()
n = int(input())
for i in range(n):
    t = int(input())
    pq.put(t)
res = 0
while len(pq.queue) > 1:
    a = pq.get()
    b = pq.get()
    res += a + b
    pq.put(a + b)
print(res)